File: sample-bots/csharp/SpinBot/SpinBot.cs
using Robocode.TankRoyale.BotApi;
using Robocode.TankRoyale.BotApi.Events;
using Robocode.TankRoyale.BotApi.Graphics;

using System.IO;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Configuration.Json;

// ------------------------------------------------------------------
// SpinBot
// ------------------------------------------------------------------
// A sample bot original made for Robocode by Mathew Nelson.
//
// Continuously moves in a circle while firing at maximum power when
// detecting enemies.
// ------------------------------------------------------------------
public class SpinBot : Bot
{
    // The main method starts our bot
    static void Main(string[] args)
    {
        new SpinBot().Start();
    }

    // Called when a new round is started -> initialize and do some movement
    public override void Run()
    {
        BodyColor = Color.Blue;
        TurretColor = Color.Blue;
        RadarColor = Color.Black;
        ScanColor = Color.Yellow;

        // Repeat while the bot is running
        while (IsRunning)
        {
            // Tell the game that when we take move, we'll also want to turn right... a lot
            SetTurnRight(10_000);
            // Limit our speed to 5
            MaxSpeed = 5;
            // Start moving (and turning)
            Forward(10_000);
        }
    }

    // We scanned another bot -> fire hard!
    public override void OnScannedBot(ScannedBotEvent evt)
    {
        Fire(3);
    }

    // We hit another bot -> if it's our fault, we'll stop turning and moving,
    // so we need to turn again to keep spinning.
    public override void OnHitBot(HitBotEvent e)
    {
        var bearing = BearingTo(e.X, e.Y);
        if (bearing > -10 && bearing < 10)
        {
            Fire(3);
        }
        if (e.IsRammed)
        {
            TurnRight(10);
        }
    }
}

File: sample-bots/python/SpinBot/SpinBot.py
from robocode_tank_royale.bot_api.bot import Bot
from robocode_tank_royale.bot_api.color import Color
from robocode_tank_royale.bot_api.events import ScannedBotEvent, HitBotEvent


# ------------------------------------------------------------------
# SpinBot
# ------------------------------------------------------------------
# A sample bot originally made for Robocode by Mathew Nelson.
#
# Continuously moves in a circle while firing at maximum power when
# detecting enemies.
# ------------------------------------------------------------------
class SpinBot(Bot):
    def run(self) -> None:
        # Set colors
        self.body_color = Color.BLUE
        self.turret_color = Color.BLUE
        self.radar_color = Color.BLACK
        self.scan_color = Color.YELLOW

        # Repeat while the bot is running
        while self.running:
            # Tell the game that when we move, we'll also want to turn right... a lot
            self.set_turn_right(10_000)
            # Limit our speed to 5
            self.max_speed = 5
            # Start moving (and turning)
            self.forward(10_000)

    def on_scanned_bot(self, e: ScannedBotEvent) -> None:
        del e
        # Fire hard!
        self.fire(3)

    def on_hit_bot(self, e: HitBotEvent) -> None:
        # If the bot is roughly in front of us, fire hard
        bearing = self.calc_bearing(self.direction_to(float(e.x), float(e.y)))
        if -10 < bearing < 10:
            self.fire(3)
        # If we rammed it, nudge to keep spinning
        if e.rammed:
            self.turn_right(10)


def main() -> None:
    bot = SpinBot()
    bot.start()


if __name__ == "__main__":
    main()
